src/codegen/codegen_aocl.cc

```diff
@@ -33,6 +33,8 @@ runtime::Module BuildAOCL(Array<LoweredFunc> funcs, std::string target_str,
 
   // Compile the .cl file.
   std::string cmd = "aoc aocl.cl";
+  // AOCL supports fp64.
+  cmd += " -Dcl_khr_fp64";
```

Comment:
It seems that the other opencl don't use this option(-Dcl_khr_fp64). Why aocl_codegen need it?
In addition, this always enables cl_khr_fp64 pragma for any fpga boards. There is no problem?

Reply:
that's a good question - what is the implication of using fp64 here? does it infer a 64bit operations? this could end up being quite costly on FPGAs no?

Reply:
The option is necessary to pass some unittests which use a float64 data type.  Although AOCL emulation can handle fp64, the AOCL compiler doesn't set the macro cl_khr_fp64.  I'm not sure why.

I think the "-Dcl_khr_fp64" option doesn't affect the compiled bitstream if we don't use float64 for the data type explicitly.

Reply:
OK, I understand it now. Thanks.